### PR TITLE
frontend tensorflow.nn convolution

### PR DESCRIPTION
**Pull request by hamehrabi**

### CODE REVIEW

**Comment by hamehrabi** [comment]
#6980

**Comment by hirwa-nshuti** [comment]
close #6736

**Comment by hamehrabi** [comment]
Hi @Auxia,

I modified some of my PR issues, but some remained due to the recent refactoring you are working on. 

These functions still have problems passing the tests:
"ivy/ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py"

    test_max_pool1d
    test_avg_pool1d
    test_avg_pool3d
    test_dct

Thanks 😊

**Comment by hamehrabi** [comment]
Hi @Auxia, I'd appreciate it if you checked and responded to my messages. It passed 2 weeks 😁😉

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

File: ivy/functional/ivy/layers.py [hunk 1]
```diff
@@ -1730,3 +1732,97 @@ def get_x_data_format(dims: int = 2, data_format: str = "channel_first"):
...
+        dims = 3
+        data_format = "channel_last"
+        return dims, data_format
+    elif data_format == "NCDHW":
+        dims = 3
+        data_format = "channel_first"
+        return dims, data_format
+
+
+@to_native_arrays_and_back
```
Comment (ahmedo42) [inline ivy/functional/ivy/layers.py:1764]: this is not related to the `convolution` function

**Review by ahmedo42 — CHANGES_REQUESTED** [review]
Hi @hamehrabi, sorry for the late review, this PR was assigned to someone who has left the team, I left a review in case you're still working on it

**Comment by ahmedo42** [comment]
Hi @hamehrabi, Thank you very much for your detailed explanation, seems you have dived deep into this :)

Any changes you make to the functional API should be made in a separate PR other than this one, we try to unify the API of all frameworks so probably a decision was made to include/exclude certain arguments, see [this section](https://lets-unify.ai/ivy/deep_dive/superset_behaviour.html)

You can definitely use a combination of other ivy functions if you sense that `conv_general_dilated()` doesn't support all requirements of the frontend.

if the `conv_general_dilated()` function is not passing the tests, then this is not your fault (since your function depends on it to work) and we can merge anyway.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
